### ISSUE TITLE
Mapping but not store

### ISSUE THREAD

**Issue by vishaldeshai** [Yalantis/FastEasyMapping#104]
Hello,

I have Core Date entity and store array of data but now i want to create array of data with the use of FastEasyMapping but dont want to store inside DB.

Thanks in advance.

**Comment by dimazen** [comment]
Hello, @vishaldeshai 
Sorry for a delay. 

FEM currently supports only mapping to either NSManagedObject or to the NSObject. Therefore it looks like that you're looking for a NSObject mapping.
In this case create your mapping by `[[FEMMapping alloc] initWithObjectClass:[MyNSObject class]]`

**Comment by vishaldeshai** [comment]
Suppose i have array of dictionary then??

**Comment by dimazen** [comment]
@vishaldeshai are you trying to map an array of dictionaries to the NSManagedObject's property?

**Comment by vishaldeshai** [comment]
I have array of dictionary and i got that from web service but i dont want to store that on Core data but only need to mapp is it possible???

**Comment by dimazen** [comment]
Sure. You may want to map them to the plain nsobject's that you have.

**Comment by vishaldeshai** [comment]
NSObject is super class for either nsaray or nsdictionary.

so we need to iterate everydictionary and map to FEMMaping right??

**Comment by dimazen** [comment]
Nope, you just need to map that exactly as you would do this for the CoreData except a different initializer for the Mapping. 

Can you share sample json and a desired output?

**Comment by vishaldeshai** [comment]
I have two types of Visit Like Plan and Unplan.

i am storing and mapping Plan Visit after Login but i dont have to store unplan Visit for my application. Json for Plan and plan visits are same.

So what is the desirable solution??

**Comment by dimazen** [comment]
You can either provide an NSObject duplicate for the Visit or to store them in the in-memory / nested managed object context that should not be saved to the database.

**Comment by vishaldeshai** [comment]
Can you elaborate more??

Json are same means format are same but values are different. :)

**Comment by dimazen** [comment]
I can't get what you mean. If the format is the same, than you should be able to map them to the same entity Visit. What I'm proposing is remove saving of the context after mapping. in this case you can get all of the results in memory. Another way is to provide NSObject's subclass with the same properties that you have for the Visit and map json to it.

**Comment by vishaldeshai** [comment]
I have already created Coredata ManagedObject Subclass.

**Comment by dimazen** [comment]
I've made a request and can not understand what is your issue. Do you want to prevent to store response in the database?

**Comment by vishaldeshai** [comment]
Yes

**Comment by dimazen** [comment]
FEM doesn't sort response. However, you can use Ordered relationship - in this way CoreData will preserve order of the response.

**Comment by dimazen** [comment]
As I already said you should either save objects into the nested context without saving or to provide class with duplicated set of the properties, that you have on the CoreData's model. In this way you'll get an array of NSObject's subclasses.

**Comment by vishaldeshai** [comment]
nested context means how??

**Comment by dimazen** [comment]
You can read it here:
https://www.cocoanetics.com/2012/07/multi-context-coredata/

In short nested context will still be used in the same way as using plain managed object context but not saving it. 

One sec:

```
FEMMapping *mapping = // your mapping here
FEMManagedObjectStore *store = [[FEMManagedObjectStore alloc] initWithContext:yourManagedObjectContext];
store.saveContextOnCommit = NO;
FEMDeserializer *deserializer = [[FEMDeserializer alloc] initWithStore:store];

// map your collection / object here. 
```

With the code above FEM will map, but won't save result to the database. i.e. `[context save:&error];` won't get called.

**Comment by vishaldeshai** [comment]
where to use mapping object in your code??

**Comment by dimazen** [comment]
@vishaldeshai I've updated my comment, please check it out.

**Comment by vishaldeshai** [comment]
saveContextOnCommit saves the object in coredata
how to prevent that?

**Comment by dimazen** [comment]
@vishaldeshai hello. 
You have to configure an underlying `FEMManagedObjectStore` by instantiating it manually and setting [saveContextOnCommit](https://github.com/Yalantis/FastEasyMapping/blob/master/FastEasyMapping/Source/Store/FEMManagedObjectStore.h#L18) to `NO`.

**Comment by vishaldeshai** [comment]
Hello,
FEMMapping *mapping = [_PlannVisit defaultMapping];// your mapping here
                            FEMManagedObjectStore *store = [[FEMManagedObjectStore alloc] initWithContext:[_PlannVisit getContext]];
                            store.saveContextOnCommit = NO;
                            FEMDeserializer *deserializer = [[FEMDeserializer alloc] initWithStore:store];
                            if (ary){
                                NSArray *objects = [deserializer collectionFromRepresentation:ary mapping:mapping];
                                self.aVisitsFlwups = [[NSMutableArray alloc] initWithArray:objects];
                            }
Here is my code but it stores in  db why??

**Comment by dimazen** [comment]
How are you fetching those objects? Most likely that they're temporary. Try to close your app right after `[deserializer collectionFromRepresentation:ary mapping:mapping];` to see that your objects was in memory.

**Comment by vishaldeshai** [comment]
that objects are feting from server.

**Comment by dimazen** [comment]
I meant how did you figure out that objects are in the database?

**Comment by vishaldeshai** [comment]
For me, there is two scenario like i am store some objects in db and some objects are not mapping but not store.

once i ma fetching stored object then also got mapped object(i.e i mensioned above)

**Comment by dimazen** [comment]
@vishaldeshai can you show the output of the debugger command:
```
po [[objects first] objectID]
```

Please, run it on a clean install.

**Comment by rahulsoni25** [comment]
@vishaldeshai
If you want to create array of core data objects without storing in db and want to use in app then try to create core data entity with [NSManagedObjectContext MR_defaultContext]
i think it will help you
